File: day1/part1.c
#include <stdio.h>
#include <stdlib.h>

// Comparison function
int compare(const void *a, const void *b)
{
    return (*(int *)a - *(int *)b);
}

int main(void)
{
    int arr1[1000];
    int arr2[1000];
    int total = 0;

    FILE *file = fopen("input1.txt", "r");

    int n = 0;
    for (int i = 0; i < sizeof(arr1) / sizeof(arr1[0]); i++)
    {
        fscanf(file, "%d %d", &arr1[i], &arr2[i]);
        n++;
    }

    qsort(arr1, n, sizeof(int), compare);
    qsort(arr2, n, sizeof(int), compare);

    for (int i = 0; i < n; i++)
    {
        total += abs(arr1[i] - arr2[i]);
    }

    printf("Total: %d ", total);
    return 0;
}

File: day1/part1.py
list1 = []
list2 = []
total = 0

with open('input1.txt', 'r') as file:
    for line in file:
        id1, id2 = line.split()
        list1.append(id1)
        list2.append(id2)
        
list1.sort()
list2.sort()

for i in range (len(list1)):
    total += abs(int(list1[i]) - int(list2[i]))

print(total)
    